packages/editor/src/editor/sidebar/fields/InputFieldWithBrowser.tsx

```diff
@@ -16,12 +17,44 @@ export const InputFieldWithBrowser = ({
   options
 }: InputFieldProps & { browsers: Array<BrowserType>; options?: TextBrowserFieldOptions }) => {
   const [open, setOpen] = useState(false);
+  const [savedSelection, setSavedSelection] = useState<{ start: number; end: number } | null>(null);
+
   const inputRef = useRef<HTMLInputElement>(null);
+  const handleTextSelection = () => {
+    if (inputRef.current) {
+      const selectionStart = inputRef.current.selectionStart ?? 0;
+      const selectionEnd = inputRef.current.selectionEnd ?? 0;
+      if (selectionStart !== selectionEnd) {
+        setSavedSelection({ start: selectionStart, end: selectionEnd });
+      } else {
+        setSavedSelection(null);
+      }
+    }
+  };
+
+  const getSelectedText = () => {
+    if (inputRef.current && savedSelection) {
+      const text = inputRef.current.value.substring(savedSelection.start, savedSelection.end);
+      return text;
+    }
+    return '';
+  };
+
   return (
     <Dialog open={open} onOpenChange={setOpen}>
       <BasicField label={label} message={message} style={{ flex: '1' }}>
         <InputGroup>
-          <Input ref={inputRef} value={value} onChange={e => onChange(e.target.value)} onBlur={onBlur} placeholder={options?.placeholder} />
+          <Input
+            ref={inputRef}
+            value={value}
+            onChange={e => onChange(e.target.value)}
+            onSelect={handleTextSelection}
+            onBlur={onBlur}
+            placeholder={options?.placeholder}
+          />
+          {browsers.some(b => b === 'CMS') && inputRef.current?.selectionStart !== inputRef.current?.selectionEnd && savedSelection && (
```

Comment:
Maybe I should add another validation, if the selection is within an EL expression, it should not be possible to execute the quick actions, otherwise EL expressions are created in EL expressions.

Reply:
I would leave it as it is for now 👍